perceval/runtime/job_group.py

```diff
@@ -413,8 +400,85 @@ def _recreate_unsent_remote_job(job_entry):
         return RemoteJob(request_data=job_entry['body'], rpc_handler=rpc_handler,
                          job_name=job_entry['body']['job_name'])
 
-    def list_unsent_jobs(self) -> list[RemoteJob]:
+    def _launch_jobs_sequential(self, delay: int, rerun: bool):
+        """
+        Launches or reruns jobs in the group on Cloud in a sequential manner.
+
+        :param delay: number of seconds to wait between the launch of to jobs on cloud
+        :param rerun: if True rerun failed jobs or run unsent jobs
+        """
+        for job_entry in self._group_info['job_group_data']:
+            if not rerun and job_entry['status'] is None:
+                remote_job = self._recreate_unsent_remote_job(job_entry)
+                remote_job.execute_async()
+            elif rerun and job_entry['status'] in ['ERROR', 'CANCELED']:
+                remote_job = self._recreate_remote_job_from_stored_data(job_entry).rerun()
+            else:
+                continue
+
+            job_entry['id'] = remote_job.id
+            job_entry['status'] = remote_job.status()
+
+            while not remote_job.is_complete:
+                time.sleep(1)
+
+            if job_entry != self._group_info['job_group_data'][-1]:
+                time.sleep(delay)  # add delay before launching next job
+        self._write_job_group_to_disk()  # update job entries on disk
+
+    def run_sequential(self, delay: int):
+        """
+        Launches the unsent jobs in the group on Cloud in a sequential manner with a
+        user-specified delay between the completion of one job and the start of the next.
+
+        :param delay: number of seconds to wait between launching jobs on cloud
+        """
+        self._launch_jobs_sequential(delay=delay, rerun=False)
+
+    def rerun_failed_sequential(self, delay: int):
+        """
+        Reruns Failed jobs in the group on the Cloud in a sequential manner with a
+        user-specified delay between the completion of one job and the start of the next.
+
+        :param delay: number of seconds to wait between re-launching jobs on cloud
+        """
+        self._launch_jobs_sequential(delay=delay, rerun=True)
+
+    def _launch_jobs_parallel(self, rerun: bool):
```

Comment:
This function can be merged with the sequential launch with just a parameter that indicates that it should wait for the job to be complete.